lib/screens/vendor_data/fetch.ex

```diff
@@ -4,13 +4,13 @@ defmodule Screens.VendorData.Fetch do
   require Logger
 
   def make_and_parse_request(url, headers \\ [], parse_fn, vendor_name, opts) do
-    with {:http_request, {:ok, response}} <- {:http_request, HTTPoison.get(url, headers, opts)},
+    with {:http_request, {:ok, response}} <- {:http_request, Mojito.get(url, headers, opts)},
          {:response_success, %{status_code: 200, body: body}} <- {:response_success, response},
          {:parse, {:ok, parsed}} <- {:parse, parse_fn.(body)} do
       {:ok, parsed}
     else
       {:http_request, {:error, e}} ->
-        log_fetch_error(vendor_name, :http_fetch_error, %{message: HTTPoison.Error.message(e)})
+        log_fetch_error(vendor_name, :http_fetch_error, %{message: e.message, reason: e.reason})
```

Comment:
It looks like the [spec](https://hexdocs.pm/mojito/Mojito.Base.html#t:error/0) for `e.reason` is `any()` (not sure what it is in practice). Maybe `inspect()` it either here or in the `log_fetch_error` function?